core/templates/components/state-editor/state-editor-properties-services/state-editor.service.spec.ts

```diff
@@ -145,5 +231,314 @@ describe('Editor state service', () => {
+    expect(ecs.correctnessFeedbackEnabled).toBe(null);
+    ecs.correctnessFeedbackEnabled = true;
+    expect(ecs.correctnessFeedbackEnabled).toBe(true);
+    ecs.correctnessFeedbackEnabled = false;
+    expect(ecs.correctnessFeedbackEnabled).toBe(false);
+  });
+
+  it('should update state content editor initialised to true', () => {
+    expect(ecs.stateContentEditorInitialised).toBe(false);
+    ecs.updateStateContentEditorInitialised();
+    expect(ecs.stateContentEditorInitialised).toBe(true);
+  });
+
+  it('should update state interaction editor initialised to true', () => {
+    expect(ecs.stateInteractionEditorInitialised).toBe(false);
+    ecs.updateStateInteractionEditorInitialised();
+    expect(ecs.stateInteractionEditorInitialised).toBe(true);
+  });
+
+  it('should update state responses initialised to true', () => {
+    expect(ecs.stateResponsesInitialised).toBe(false);
+    ecs.updateStateResponsesInitialised();
+    expect(ecs.stateResponsesInitialised).toBe(true);
+  });
+
+  it('should update state hints editor initialised to true', () => {
+    expect(ecs.stateHintsEditorInitialised).toBe(false);
+    ecs.updateStateHintsEditorInitialised();
+    expect(ecs.stateHintsEditorInitialised).toBe(true);
+  });
+
+  it('should update state solution editor initialised to true', () => {
+    expect(ecs.stateSolutionEditorInitialised).toBe(false);
+    ecs.updateStateSolutionEditorInitialised();
+    expect(ecs.stateSolutionEditorInitialised).toBe(true);
+  });
+
+  it('should update state editor directive initialised to true', () => {
+    expect(ecs.stateEditorDirectiveInitialised).toBe(false);
+    ecs.updateStateEditorDirectiveInitialised();
+    expect(ecs.stateEditorDirectiveInitialised).toBe(true);
+  });
+
+  it('should update update current rule input is valid', () => {
+    expect(ecs.checkCurrentRuleInputIsValid()).toBe(false);
+    ecs.updateCurrentRuleInputIsValid(true);
+    expect(ecs.checkCurrentRuleInputIsValid()).toBe(true);
+    ecs.updateCurrentRuleInputIsValid(false);
+    expect(ecs.checkCurrentRuleInputIsValid()).toBe(false);
+  });
+
+  it('should set state names and emit state names change event', () => {
+    expect(ecs.getStateNames()).toEqual([]);
+    ecs.setStateNames(['Introduction', 'State1']);
+    expect(ecs.getStateNames()).toEqual(['Introduction', 'State1']);
+    ecs.setStateNames(['Introduction', 'End']);
+    expect(ecs.getStateNames()).toEqual(['Introduction', 'End']);
+  });
+
+  it('should check event listener registration status', () => {
+    expect(ecs.checkEventListenerRegistrationStatus()).toBe(false);
+    ecs.updateStateInteractionEditorInitialised();
+    expect(ecs.checkEventListenerRegistrationStatus()).toBe(false);
+    ecs.updateStateResponsesInitialised();
+    expect(ecs.checkEventListenerRegistrationStatus()).toBe(false);
+    ecs.updateStateEditorDirectiveInitialised();
+    expect(ecs.checkEventListenerRegistrationStatus()).toBe(true);
```

Comment:
Can we add a note here why the registration status is updating only after calling `updateStateEditorDirectiveInitialised`?

Reply:
Done